core/src/apps/cardano/helpers/chunks.py

```diff
@@ -0,0 +1,83 @@
+from micropython import const
+from typing import TYPE_CHECKING
+
+from trezor import messages, protobuf
+from trezor.wire import ProcessError
+from trezor.wire.context import call as ctx_call
+
+if TYPE_CHECKING:
+    from typing import Generic, TypeVar, Union
+
+    Chunk = Union[
+        messages.CardanoTxInlineDatumChunk,
+        messages.CardanoTxReferenceScriptChunk,
+        messages.CardanoMessagePayloadChunk,
+    ]
+
+    C = TypeVar("C", bound=Chunk)
+else:
+    # typechecker cheat
+    Generic = (object,)
+    C = Chunk = 0
+
+MAX_CHUNK_SIZE = const(1024)
+
+
+def _get_chunks_count(size: int) -> int:
+    """Integer-only version of `ceil(size / MAX_CHUNK_SIZE)`."""
+    assert size >= 0
+    return 0 if size == 0 else (size - 1) // MAX_CHUNK_SIZE + 1
+
+
+def _validate_chunk(
+    chunk: Chunk,
+    chunk_index: int,
+    total_size: int,
+) -> None:
+    chunks_count = _get_chunks_count(total_size)
+    assert chunk_index < chunks_count
+
+    chunk_name = chunk.__class__.__name__
+
+    if len(chunk.data) > MAX_CHUNK_SIZE:
+        raise ProcessError(f"Invalid {chunk_name}: Too large")
+
+    is_last_chunk = chunk_index == chunks_count - 1
+
+    if not is_last_chunk and len(chunk.data) < MAX_CHUNK_SIZE:
```

Comment:
Maybe checking this is not really needed? (Not sure e.g. because of UI.)
(We are not this strict on Ledger.)

Reply:
Maybe we could do without the check, but since the communication is usually (always?) done by Connect, the client is not affected by the check in any way. And having a standard form is good security-wise, I think.